lib/ecto/changeset/relation.ex

```diff
@@ -252,6 +251,27 @@ defmodule Ecto.Changeset.Relation do
     Changeset.change(current, changes) |> put_new_action(:update)
   end
 
+  @doc """
+  TODO
+  """
+  def on_replace(%{on_replace: :mark_as_invalid}, changeset_or_model) do
+    %{Changeset.change(changeset_or_model) | valid?: false}
```

Comment:
I don't think we should do this. We probably should mark the whole field as invalid by triggering this clause: https://github.com/elixir-lang/ecto/blob/master/lib/ecto/changeset.ex#L424. This way, we will at least show an error message to the user instead of silently failing.